### DISCUSSION TITLE
OpenSergo 是基于k8s 做的云原生扩展吗？

### DISCUSSION THREAD

**Discussion by su-yh** [opensergo/opensergo-specification#82] [Q&A]
OpenSergo 是基于k8s 做的云原生扩展吗？

对于普通 的spring boot 应用是不是不太适合用这个opensergo 。

因为我看这里面有很多k8s 的影子，比如说一些yaml 配置。还有一些术语啥的？

另外 就是流量控制和路由，这些都不是普通的sdk 能处理的，有点像是基于envoy 的？

**Comment by sczyh30** [comment]
> OpenSergo 是基于k8s 做的云原生扩展吗？对于普通 的spring boot 应用是不是不太适合用这个opensergo 。

其实这些服务治理能力是无关于应用架构的，OpenSergo 的终态设计目标是同时支持 K8s 和非 K8s 体系；从目前 OpenSergo 控制面实现情况来说，目前控制面暂时贴近 K8s 环境部署，在近几个月内 社区将重构控制面、支持非 K8s 体系。

> 另外 就是流量控制和路由，这些都不是普通的sdk 能处理的，有点像是基于envoy 的？

流量控制和路由的原子能力 都是可以沉淀到 SDK 侧（即 Sentinel 2.0），但一般需要配合具体的框架使用（如 Dubbo、Spring Cloud、gRPC 适配）；Envoy 我们也会结合 Istio 生态支持。

**Reply by su-yh** [reply]
好的，了解 了。谢谢回复。^_^

**Reply by su-yh** [reply]
另外，opensergo 是以sidecar 模式存在的，还是以sdk 的形式存在的呢？
